api/src/main/java/com/epam/pipeline/entity/cluster/CorePodInstance.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2017-2024 EPAM Systems, Inc. (https://www.epam.com/)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.epam.pipeline.entity.cluster;
+
+import io.fabric8.kubernetes.api.model.Pod;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+
+@EqualsAndHashCode(callSuper = true)
+@Data
+public class CorePodInstance extends PodInstance {
+
+    private String parentType;
+    private String parentName;
+    private PodInstanceStatus status;
```

Comment:
What do you think if we will add pod status and containers to the regular `PodInstance` class? do you see any problems with that approach? I think it is reasonable to have these fields not only for `CorePod`